### PR TITLE
bump rust to 1.56.1

### PR DESCRIPTION
**Pull request by pozsgaic**
Signed-off-by: pozsgaic <chris.pozsgai@progress.com>

### CODE REVIEW

File: support/allowed_lints.txt [hunk 1]
```diff
@@ -1,3 +1,5 @@
 clippy::module_inception
 clippy::new_ret_no_self
 clippy::new_without_default
+clippy::suspicious_else_formatting
+unused_must_use
```
Comment (mwrock) [inline support/allowed_lints.txt:5]: I think you just need to put these allows in https://github.com/davidMcneil/configopt/blob/master/configopt-derive/src/configopt_type/generate.rs and then fork that. That will isolate the exceptions to only ConfigOpt generated code.

**Review by mwrock — CHANGES_REQUESTED** [review]

File: support/allowed_lints.txt (hunk 1 above)
Comment (pozsgaic) [reply support/allowed_lints.txt:5]: We will need to update our habitat-sh/structopt code as well to fix the clippy::suspicious_else_formatting linter error

File: components/hab/Cargo.toml [hunk 2]
```diff
@@ -17,7 +17,7 @@ base64 = "*"
 bitflags = "*"
 chrono = {version = "*", features = ["serde"]}
 clap = { git = "https://github.com/habitat-sh/clap.git", branch = "v2-master", features = [ "suggestions", "color", "unstable" ] }
-configopt = { git = "https://github.com/davidMcneil/configopt.git" }
+configopt = { git = "https://github.com/pozsgaic/configopt.git" }
```
Comment (mwrock) [inline components/hab/Cargo.toml:20]: we should fork this into the habitat-sh org.

**Review by mwrock — CHANGES_REQUESTED** [review]

File: components/hab/Cargo.toml (hunk 2 above)
Comment (pozsgaic) [reply components/hab/Cargo.toml]: I forked to habitat-sh and renamed master to main in the process

**Review by mwrock — CHANGES_REQUESTED** [review]
We want to remove changes to VERSION and CHANGELOG.md since those should only be auto generated by expeditor.

**Comment by pozsgaic** [comment]
I don't recall adding this but maybe I pulled it in merging from main?  Will fix.

**Review by mwrock — APPROVED** [review]

**Pull request merged**